Anature/src/application/controllers/BattleController.java

```diff
@@ -1117,50 +1194,57 @@ public void run()
 						mDialogueTxt.set("You clicked on Escape!\nThat has yet to be implemented!");
 						mCanClick.set(true);
 					}
+
 				});
 				break;
 
 			case Switch:
-				mClickQueue.enqueue(new Runnable()
+				activateSwitch();
+				break;
+		}
+	}
+
+	private void activateSwitch()
+	{
+		mClickQueue.enqueue(new Runnable()
+		{
+			@Override
+			public void run()
+			{
+				mFightManager.setPlayerSelectedIndex(mSwitchIndexSelected);
+				Anature oldAnature = mPlayer.getAnatures().get(mPlayer.getSelectedIndex());
+				mPlayer.setSelectedIndex(mSwitchIndexSelected);
+				Anature newAnature = mPlayer.getAnatures().get(mPlayer.getSelectedIndex());
+
+				OpacityAnimation fadeOld = new OpacityAnimation(mAnatureBack, Duration.millis(400), false);
+				fadeOld.setOnFinished(new EventHandler<ActionEvent>()
 				{
 					@Override
-					public void run()
+					public void handle(ActionEvent event)
 					{
-						mFightManager.setPlayerSelectedIndex(mSwitchIndexSelected);
-						Anature oldAnature = mPlayer.getAnatures().get(mPlayer.getSelectedIndex());
-						mPlayer.setSelectedIndex(mSwitchIndexSelected);
-						Anature newAnature = mPlayer.getAnatures().get(mPlayer.getSelectedIndex());
+						updatePlayerAnature(newAnature);
 
-						OpacityAnimation fadeOld = new OpacityAnimation(mAnatureBack, Duration.millis(400), false);
-						fadeOld.setOnFinished(new EventHandler<ActionEvent>()
+						try
 						{
-							@Override
-							public void handle(ActionEvent event)
-							{
-								updatePlayerAnature(newAnature);
-
-								try
-								{
-									Thread.sleep(500);
-								}
-
-								catch(InterruptedException e)
-								{
-									LoggerController.logEvent(LoggingTypes.Error, e.getMessage());
-								}
-
-								OpacityAnimation fadeInNew = new OpacityAnimation(mAnatureBack, Duration.millis(400), true);
-								fadeInNew.setOnFinished(actionEvent -> mCanClick.set(true));
-								fadeInNew.play();
-							}
-						});
-
-						fadeOld.play();
-						mDialogueTxt.set("Come on back " + oldAnature.getName() + ".");
+							Thread.sleep(500);
+						}
+
+						catch(InterruptedException e)
+						{
+							LoggerController.logEvent(LoggingTypes.Default, e.getMessage());
```

Comment:
We changed it so that the "Default" enum is now "Error", so you will have to change that.